sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,445 @@ def reduce_inequalities(inequalities, symbols=[]):
+        piv_rows.append(k)
+        i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    # Phase 2: starting at a feasible solution, pivot until we reach optimal solution
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+
+        # Choose a pivot column, j0
+        piv_cols = []
+        piv_cols = [j for j in range(C.cols) if C[j] < 0]
+        if not piv_cols:
+            break
+        j0 = sorted(piv_cols, key=lambda c: R[c])[0] # Bland's rule
+
+        # Choose a pivot row, i0
+        piv_rows = [i for i in range(A.rows) if A[i, j0] > 0]
+        if not piv_rows:
+            raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large values!')
+        i0 = _choose_pivot_row(A, B, piv_rows, j0, S)
+
+        M = _pivot(M, i0, j0)
+        R[j0], S[i0] = S[i0], R[j0]
+
+    argmax = [None]*(M.cols-1)
+    argmin_dual = [None]*(M.rows-1)
+
+    for i, var in enumerate(R):
+        v = var[0]
+        n = var[1]
+        if v == False:
+            argmax[n] = 0
+        else:
+            argmin_dual[n] = M[-1, i]
+
+    for i, var in enumerate(S):
+        v = var[0]
+        n = var[1]
+        if v == True:
+            argmin_dual[n] = 0
+        else:
+            argmax[n] = M[i, -1]
+
+    return M[-1, -1], argmax, argmin_dual
+
+
+def linprog_from_matrices(A, B, C):
+    """
+    Because of the duality of linear programming, there are two valid ways to
+    interpret this function. The first is as a solver for the standard
+    maximization problem:
+        Maximizing Cx constrained to Ax <= B and x >= 0.
+    The second as is a solver for the standard minimization problem:
+        Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    x is a column vector of variables, and y is a column vector of dual
+    variables.
+
+    If we keep the maximization interpretation in mind, then matrix A and
+    column vector B represent a set of constraints. For example,
+        1*x + 0*y <= 2
+        0*x + 0*y <= 4
+    can be represented as:
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> A = Matrix([[1, 0],
+    ...             [0, 1]])
+    >>> B = Matrix([[2],
+    ...             [4]])
+
+    In this interpretation, row vector C is an objective function to be
+    maximized. For example,
+        3x + y
+    can be represented as:
+
+    >>> C = Matrix([[3, 1]])
+
+    Parameters
+    ==========
+
+    Under the maximization interpretation, m is the number of constraints and n
+    is the number of variables.
+
+    A : Matrix with shape (m, n)
+        Must contain only Rational and float coefficients.
+
+    B : Matrix with shape (m, 1)
+        Must contain only Rational and float coefficients.
+
+    C: Matrix with shape (1, n)
+        Must contain only Rational and float coefficients.
+
+    Returns
+    =======
+
+    Returns `(optimum, argmax, and argmax_dual)`:
```

Comment:
Per comment on main page, I think `argmax_dual` should be `argmin_dual`. And optimum should be noted as the max value for the objective and the min value for the dual objective.

Reply:
Yeah I agree.

Reply:
Also, looking at the `LP` code [here](https://github.com/sympy/sympy/pull/22389#issuecomment-1211097848) I think the slack variables are needed when working with `unbound` (or not nonnegative) values for variables.

Reply:
I also like the idea of immediately getting rid of equalities by substitution rather than creating another inequality to solve.

Reply:
> I think the slack variables are needed when working with unbound (or not nonnegative) values for variables.

I just tested thousands of randomly generated problems and everything seems to be working. Numbers for the tableau  can be positive or negative or zero.

> I also like the idea of immediately getting rid of equalities by substitution rather than creating another inequality to solve.

What advantage would this have?

Reply:
> I think the slack variables are needed when working with unbound (or not nonnegative) values for variables.

oh I think I misunderstood what you meant. LP problems in standard form have a non-negativity constraint on all variables but currently the function doesn't impose such a constraint.

Reply:
> What advantage would this have?

Given c constraints, m of which are written as equalities, instead of creating c + m constraints to solve you will have c - m to solve after solving for m of the symbols in the equalities. So if there were 10 contraints and 5 involved equalities, you would end up with 5 contraints instead of 15. Maybe the time resolving the contraints is comparable to the time needed to find the optimum. I am not sure.

Reply:
> have a non-negativity constraint

So is *that* the reason for the slack variables: making sure the quantity is not negative?

Reply:
> Maybe the time resolving the contraints is comparable to the time needed to find the optimum. I am not sure.

I suspect the time would be similar, though I'm not sure either. It would also change the number dual variables which might matter.

>  have a non-negativity constraint

I was wrong about this. They do have a non-negativity constraint. I'm not sure why I believed this, but it sure did make me lose a lot of time.